__tests__/app.test.js

```diff
@@ -195,21 +196,69 @@ describe("GET /api/articles/:article_id/comments", () => {
     return request(app)
       .get("/api/articles/xyz/comment")
       .expect(400)
       .then(({ body }) => {
         expect(body.msg).toEqual("Bad request. Please check what you're requesting and try again.");
       });
-    });
+  });
+
+  test("404: responds with JSON object of all /api/articles/:article_id/comments endpoints", () => {
+    return request(app)
+      .get("/api/articles/1500/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Not Found");
+      });
+  });
+});
+
+//07
+describe("POST /api/articles/:article_id/comments", () => {
+  test("200: responds with a JSON object of the comments of a given article id Which has been posted.", () => {
+    const newComments = {username: 'lurker', body: 'Comments...'};
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComments)
+      .expect(201)
+      .then(({ body }) => {
+        const {comments} = body;
 
-    test("404: responds with JSON object of all /api/articles/:article_id/comments endpoints", () => {
+          expect(comments[0]).toHaveProperty("comment_id", expect.any(Number));
+          expect(comments[0]).toHaveProperty("votes", expect.any(Number));
+          expect(comments[0]).toHaveProperty("created_at", expect.any(String));
+          expect(comments[0]).toHaveProperty("author", expect.any(String));
+          expect(comments[0]).toHaveProperty("body", expect.any(String));
+          expect(comments[0]).toHaveProperty("article_id", expect.any(Number));
+
+          expect(comments[0]["votes"]).toEqual(1);
+          expect(comments[0]["author"]).toEqual('lurker');
+          expect(comments[0]["body"]).toEqual('Comments...');
+      });
+  });
+
+  test("400: responds with an error message if there's a malfunction request, for example missing NOT NULL value/s.", () => {
+  const newComments = {username: 1, body:null}; //Creating violation for the 'NOT NULL' column. Check errors.js for details.
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComments)
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toMatch("Malfunction: Class 23 — Integrity Constraint Violation:")
+      });
+  });
+
+  test("400: responds with an error message if there's an INCORRECT DATA TYPE in REQUEST BODY.", () => {
+  const newComments = {username: 1, body:'Comments...', votes: 'x'}; //Creating data exception error. votes should be an integer.
       return request(app)
-        .get("/api/articles/1500/comments")
-        .expect(404)
+        .post("/api/articles/1/comments")
+        .send(newComments)
+        .expect(400)
         .then(({ body }) => {
-          expect(body.msg).toEqual("Not Found");
+            expect(body.msg).toMatch("Bad request. Data exception Error")
         });
-    });
-});
\ No newline at end of file
+  });
+});
+
```

Comment:
You are missing some tests 

400 bad request with an invalid id - notanid etc 
404 not found with a non-existent id - 4000 etc 
404 not found - username does not exist in the database (psql will throw it's on error for you to utilise)

Reply:
Took me a while to get all three right, but got it now.